skillmap/src/components/AppModal.tsx

```diff
@@ -487,13 +539,34 @@ export class AppModalImpl extends React.Component<AppModalProps, AppModalState>
             )
         }
 
+        const handleInputRef = (ref: HTMLInputElement) => {
+            if (ref) this.textInput = ref;
+        }
+
+        const firstName = signedIn && profile && pxt.auth.firstName(profile);
+
         return <Modal title={title} onClose={this.handleOnClose} actions={buttons}>
             {lf("Use the button below to get your completion certificate!")}
             {reward.previewUrl &&
                 <div className="certificate-reward">
                     <img src={reward.previewUrl} alt={lf("certificate Preview")} />
                 </div>
             }
+            {isApprovedPdfCert &&
+                <Input
+                    className="cert-name-input"
+                    placeholder={"Put your name on it!"}
+                    label={lf("Enter your name")}
+                    type="text"
+                    initialValue={firstName || undefined}
+                    handleInputRef={handleInputRef}
+                    preserveValueOnBlur={true}
+                    onEnterKey={onCertificateClick}
+                    title={lf("Enter a custom name to put on the certificate")}
```

Comment:
```suggestion
                    title={lf("Enter your name to customize the certificate")}
```

"Custom name" sounded a bit weird to me, but I don't feel strongly either way :)

Reply:
Woops I meant to do a pass on the wording on these and totally forgot, thanks!